dnf5-plugins/needs_restarting_plugin/needs_restarting.cpp

```diff
@@ -456,7 +456,7 @@ void NeedsRestartingCommand::processes_need_restarting(Context & ctx, bool exclu
 
     // Map of executable paths to their process start times and packages
     struct ProcessInfo {
-        const char * pid;
+        std::string pid;
```

Comment:
I made this an owned `std::string` since the `char * entry->d_name` doesn't live long enough.